CMakeLists.txt

```
@@ -11,6 +11,7 @@ option(Idefix_MHD "enable MHD" OFF)
 option(Idefix_MPI "enable Message Passing Interface parallelisation" OFF)
 option(Idefix_HIGH_ORDER_FARGO "Force Fargo to use a PPM reconstruction scheme" OFF)
 option(Idefix_DEBUG "Enable Idefix debug features (makes the code very slow)" OFF)
+option(Idefix_RUNTIME_CHECKS "Enable runtime detection of book-keeping errors" OFF)
```

this phrasing was written with the current application (particles) in mind, but this could be more general. Maybe something like 
```suggestion
option(Idefix_RUNTIME_CHECKS "Enable runtime sanity checks" OFF)
```